_resources/port1.0/group/janet-1.0.tcl

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+# Usage:
+# PortGroup     janet 1.0
+#
+# This PortGroup allows projects to be built with jpm
+#     (see https://janet-lang.org/docs/jpm.html).
+#
+
+# search in ${prefix}/lib/janet/bin and create link in ${prefix}/bin
+options janet.link_bin
+default janet.link_bin          yes
+
+# program to create and maintain library archives
+options janet.compiler.ar
+default janet.compiler.ar       {[janetpg::get_default_ar]}
+
+namespace eval janetpg {
+}
+
+proc janetpg::get_default_ar {} {
+    # no doubt that at some point, for some systems, the ar from cctools will have to be used
+    return /usr/bin/ar
+}
+
+# The build command jpm does not respect CFLAGS/CXXFLAGS.
+# You can set the compilers, but no spaces seem to be allowed.
+# Hence, we cannot simply append the CFLAGS/CXXFLAGS to the compiler name.
+# As a workaround, create custom compilers with CFLAGS/CXXFLAGS baked in.
+pre-configure {
+    xinstall -d -m 0755 ${worksrcpath}/macports_bin
+    foreach bin {cc cxx ar} {
+        set fl [open "${worksrcpath}/macports_bin/${bin}" w 0755]
+        puts ${fl} "#!/bin/bash"
+        puts -nonewline ${fl} "exec "
+        close ${fl}
+    }
+}
+post-configure {
+    foreach bin {cc cxx ar} {
+        set fl [open "${worksrcpath}/macports_bin/${bin}" a 0755]
+        puts ${fl} { "$@"}
+        close ${fl}
+    }
+}
```

Comment:
Not sure I see any advantage to creating the wrapper scripts in three different phases... 

Wouldn’t it be simpler to just create the entire file in one go in the main configure phase ?

Reply:
I agree splitting the wrapper script generation into `pre-configure`, `configure`, and `post-configure` is less than ideal.
The problem is that `configure { ... }` means `portconfigure::configure_main` from port [portconfigure.tcl](https://github.com/macports/macports-base/blob/a0172d6655263e9de47e05dd33da0651c1b93ced/src/port1.0/portconfigure.tcl#L1640) does not run, and so none of the `append_to_environment_value configure CFLAGS/CXXFLAGS` commands are called.
I was hoping to just use $CFLAGS add defined by the MacPorts base.

Reply:
I was not suggesting overriding the entire configure phase but just create the entire wrap script with a single echo command. Line breaks can be passed as \n etc.

Reply:
Thank you for the clarification.
It would certainly require less code.
I will try to make the change tomorrow.

Reply:
Actually... you also don’t need to evaluate the environment variables when you write them to file, as you should actually write $CC into the file, so that the variable is then evaluated each time the script is  ran during the build phase.

Reply:
I did not think `$CC` (and `$CXX`) were set by the MacPorts base in the build phase.
However, I could be wrong.

Reply:
If not you could append them yourself to the build env.

Reply:
In that case, it seems just easier to write `$CC` and `$CXX` directly to the wrapper file and be done with it.
I cannot image a case where `$CC/$CXX` are different in the configure and build phases.

Reply:
Yes, but my point is it does not then matter if they are actually set or not when you create the scripts, it only matters that they are set when they are used. So you could create these scripts in any phase, e.g. post-extract or whatever.

Reply:
Cresting the script in post-extract would also have the advantage you would have to play tricks with echo etc to get them into a single ‘configure’ command, which is a bit ugly. You could just use the standard tcl puts.

Reply:
I may be missing some important part of your concern.
Please forgive me if that is the case.

It seems to me that the wrapper scripts can *only* be generated during the configure phase since that is the *only* time `$CFLAGS/$CXXFLAGS` are fully set by the MacPorts base.

Were it not for `$CFLAGS/$CXXFLAGS`, then I would agree that there would be much easier ways of generating the wrapper scripts.
Since `$CFLAGS/$CXXFLAGS` must be set during the configure phase, we might as well set `$CC/$CXX` as well and not worry about setting any more environmental variables in the build phase.

Reply:
Its not I have a huge concern, I am sure it works. It just looked a little overly complicated having the file creation split into three parts.